students/itsabhi/Lesson8/circle.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python3
+
+import math
+
+
+class Circle:
+    # Circle class
+    _diameter = None  # This cannot be stored as its not an object attribute
+    _area = None  # This cannot be stored as its not an object attribute
+
+    @property
+    def diameter(self):
+        return self.radius*2
+
+    def __init__(self, the_radius):
+        self.radius = the_radius
+        if not self.radius > 0:
+            print("Invalid radius....\n")
+
+    @diameter.setter
+    def diameter(self, value):
+        # Sets radius. Note diameter is not stored as its not an object attribute.
+        self.radius = value/2
+
+    @property
+    def area(self):
+        return (self.radius**2)*math.pi
+
+    @classmethod
+    def from_diameter(cls, diam):
+        # Returns a Circle class object
+        return cls(diam/2)
+
+    def __str__(self):
+        return("Circle with radius: "+str(self.radius))
```

Comment:
you don't need parens with return and don't forget proper spacing:
`return "Circle with radius: " + str(self.radius)`